recipes/cpprestsdk/all/conanfile.py

```diff
@@ -109,37 +107,40 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "cpprestsdk"))
 
     def package_info(self):
-        if self.settings.compiler == "Visual Studio":
-            debug_suffix = 'd' if self.settings.build_type == 'Debug' else ''
-            toolset = {'12': '120',
-                       '14': '140',
-                       '15': '141',
-                       '16': '142'}.get(str(self.settings.compiler.version))
-            version_tokens = self.version.split(".")
-            versioned_name = "cpprest%s_%s_%s%s" % (toolset, version_tokens[0], version_tokens[1], debug_suffix)
-            # CppRestSDK uses different library name depends on CMAKE_VS_PLATFORM_TOOLSET
-            if not os.path.isfile(os.path.join(self.package_folder, 'lib', '%s.lib' % versioned_name)):
-                versioned_name = "cpprest_%s_%s%s" % (version_tokens[0], version_tokens[1], debug_suffix)
-            lib_name = versioned_name
-        else:
-            lib_name = 'cpprest'
-
-        self.cpp_info.libs.append(lib_name)
+        # cpprestsdk_boost_internal
+        self.cpp_info.components["cpprestsdk_boost_internal"].includedirs = []
+        self.cpp_info.components["cpprestsdk_boost_internal"].requires = ["boost::boost"]
+        # cpprestsdk_openssl_internal
+        self.cpp_info.components["cpprestsdk_openssl_internal"].includedirs = []
+        self.cpp_info.components["cpprestsdk_openssl_internal"].requires = ["openssl::openssl"]
+        # cpprest
+        self.cpp_info.components["cpprest"].libs = tools.collect_libs(self)
+        self.cpp_info.components["cpprest"].requires = ["cpprestsdk_boost_internal", "cpprestsdk_openssl_internal"]
         if self.settings.os == "Linux":
-            self.cpp_info.system_libs.append("pthread")
+            self.cpp_info.components["cpprest"].system_libs.append("pthread")
         elif self.settings.os == "Windows":
             if self.options.get_safe("http_client_impl") == "winhttp":
-                self.cpp_info.system_libs.append("winhttp")
+                self.cpp_info.components["cpprest"].system_libs.append("winhttp")
             if self.options.get_safe("http_listener_impl") == "httpsys":
```

Comment:
should there be a component for https://github.com/microsoft/cpprestsdk/blob/319dd31f7ba8752683aaab2679a1c8fefcf48ef3/Release/src/CMakeLists.txt#L269?

Reply:
good catch, I'll fix that

Reply:
Well, after digging into CMakeLists, this component can't exist in current recipe. `http_client_impl` should be updated to add 
`winhttppal` value: https://github.com/microsoft/cpprestsdk/blob/319dd31f7ba8752683aaab2679a1c8fefcf48ef3/Release/src/CMakeLists.txt#L134
Maybe in another PR?